atmo/init_proxy.go

```diff
@@ -2,8 +2,144 @@
+	case "collector":
+		logger.Info("configuring collector exporter for tracing")
+
+		collectorOpts, err := collectorExporterOptions(config.Collector)
+		if err != nil {
+			return func() {}, errors.Wrap(err, "collectorExporterOptions")
+		}
+
+		exporterString = "collector"
+		exporterOpts = append(exporterOpts, collectorOpts...)
+
+		logger.Info("created collector trace exporter")
+	default:
+		logger.Warn(fmt.Sprintf("unrecognised tracer type configuration [%s]. Defaulting to no tracer", config.TracerType))
+		fallthrough
+	case "none", "":
+		otel.SetTracerProvider(traceProviders.NewNoopTracerProvider())
+
+		logger.Info("finished setting up noop tracer")
+
+		return func() {}, nil
+	}
+
+	exporter, err := otlptrace.New(context.Background(), otlptracegrpc.NewClient(exporterOpts...))
+	if err != nil {
+		return func() {}, errors.Wrapf(err, "oltptrace.New with exporter as %s", exporterString)
+	}
+
+	traceOpts := []trace.TracerProviderOption{
+		trace.WithSampler(trace.TraceIDRatioBased(config.Probability)),
+		trace.WithResource(
+			resource.NewWithAttributes(
+				semconv.SchemaURL,
+				semconv.ServiceNameKey.String(config.ServiceName),
+				attribute.String("exporter", exporterString),
+			),
+		),
+		trace.WithBatcher(exporter,
+			trace.WithMaxExportBatchSize(trace.DefaultMaxExportBatchSize),
+			trace.WithBatchTimeout(trace.DefaultScheduleDelay*time.Millisecond),
+			trace.WithMaxExportBatchSize(trace.DefaultMaxExportBatchSize),
+		),
+		trace.WithResource(
+			resource.NewWithAttributes(
+				semconv.SchemaURL,
+				semconv.ServiceNameKey.String(config.ServiceName),
+				attribute.String("exporter", "honeycomb"),
+			),
+		),
+	}
+
+	traceProvider := trace.NewTracerProvider(traceOpts...)
+
+	otel.SetTracerProvider(traceProvider)
+
+	logger.Info(fmt.Sprintf("finished setting up tracer [%s] with a trace probability of [%f]",
+		exporterString, config.Probability))
+
+	return func() {
+		_ = traceProvider.Shutdown(context.Background())
+	}, nil
+
+}
+
+func collectorExporterOptions(config *options.CollectorConfig) ([]otlptracegrpc.Option, error) {
+	if config == nil {
+		return nil, errors.New("empty collector tracer configuration")
+	}
+
+	ctx, ctxCancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer ctxCancel()
+
+	conn, err := grpc.DialContext(ctx, config.Endpoint, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	if err != nil {
+		return nil, errors.Wrap(err, "grpc.DialContext")
+	}
+
+	return []otlptracegrpc.Option{
+		otlptracegrpc.WithGRPCConn(conn),
+		otlptracegrpc.WithTimeout(500 * time.Millisecond),
+	}, nil
+}
+
+func honeycombExporterOptions(config *options.HoneycombConfig) ([]otlptracegrpc.Option, error) {
+	if config == nil {
+		return nil, errors.New("empty honeycomb tracer configuration")
+	}
+
+	return []otlptracegrpc.Option{
+		otlptracegrpc.WithEndpoint(config.Endpoint),
+		otlptracegrpc.WithHeaders(map[string]string{
+			"x-honeycomb-team":    config.APIKey,
```

Comment:
Odd that the APIKey goes into the `team` header? 🤔

Reply:
It is, but it's what they say we should do: https://docs.honeycomb.io/getting-data-in/opentelemetry/go/